packages/grpc-js/src/server-call.ts

```diff
@@ -559,52 +544,89 @@ export class Http2ServerCallStream<
     return metadata;
   }
 
-  receiveUnaryMessage(encoding: string): Promise<RequestType> {
-    return new Promise((resolve, reject) => {
-      const stream = this.stream;
-      const chunks: Buffer[] = [];
-      let totalLength = 0;
+  receiveUnaryMessage(
+    encoding: string,
+    next: (
+      err: Partial<ServerStatusResponse> | null,
+      request?: RequestType
+    ) => void
+  ): void {
+    const { stream } = this;
+
+    let receivedLength = 0;
+    const call = this;
+    const body: Buffer[] = [];
+    const limit = this.maxReceiveMessageSize;
+
+    stream.on('data', onData);
+    stream.on('end', onEnd);
+    stream.on('error', onEnd);
+
+    function onData(chunk: Buffer) {
+      receivedLength += chunk.length;
+
+      if (receivedLength > limit) {
```

Comment:
This doesn't handle the case where the limit is `-1`, which means "unlimited". You can see how this is handled on line 577 of the current implementation.

Reply:
yeah, already noticed the bug I've created - will add `receivedLength !== -1`

Reply:
No, it should be `limit !== -1`.

Reply:
right, of course :)